app/routes/verify.js

```diff
@@ -0,0 +1,27 @@
+import Route from '@ember/routing/route';
+import UnauthenticatedRouteMixin from 'ember-simple-auth/mixins/unauthenticated-route-mixin';
+
+export default Route.extend(UnauthenticatedRouteMixin, {
+  titleToken() {
+    return this.get('l10n').t('Verify');
+  },
+
+  beforeModel(transition) {
+    let payload = {
+      data: {
+        token: transition.queryParams.token
+      }
+    };
+
+    return this.get('loader')
+      .post('auth/verify-email', payload)
+      .then(() => {
+        this.controllerFor('verify').message = 'Your email has been verified successfully! Please login to continue.';
```

Comment:
This is going against the ember design principles. First of all, this entire logic should be an action inside the controller. 
Secondly, the approach is very hacky, you are using a route to manipulate properties of it's controller which are available readily inside the controller.
Even after shifting the property to the controller, keep all the message text inside the template itself and use `if-else` handlebar helpers to decide which message to display.

Reply:
After moving the code to the `verify` controller, how do I trigger it automatically after the verification link is clicked?